artiq/applets/plot_xy.py

```diff
@@ -24,32 +31,57 @@ def data_changed(self, data, mods, title):
         fit = data.get(self.args.fit, (False, None))[1]
 
         if not len(y) or len(y) != len(x):
-            return
+            self.mismatch['X values'] = True
+            self.timer.start(1000)
+        else:
+            self.mismatch['X values'] = False
+            if not sum(self.mismatch.values()):
+                self.timer.stop()
         if error is not None and hasattr(error, "__len__"):
             if not len(error):
                 error = None
             elif len(error) != len(y):
-                return
+                self.mismatch['Error bars'] = True
+                self.timer.start(1000)
+            else:
+                self.mismatch['Error bars'] = False
+                if not sum(self.mismatch.values()):
+                    self.timer.stop()
         if fit is not None:
             if not len(fit):
                 fit = None
             elif len(fit) != len(y):
-                return
+                self.mismatch['Fit values'] = True
+                self.timer.start(1000)
+            else:
+                self.mismatch['Fit values'] = False
+                if not sum(self.mismatch.values()):
+                    self.timer.stop()
 
         self.clear()
+        if self.mismatch['X values']:
+            return
         self.plot(x, y, pen=None, symbol="x")
         self.setTitle(title)
-        if error is not None:
+        if error is not None and not self.mismatch['Error bars']:
             # See https://github.com/pyqtgraph/pyqtgraph/issues/211
             if hasattr(error, "__len__") and not isinstance(error, np.ndarray):
                 error = np.array(error)
             errbars = pyqtgraph.ErrorBarItem(
                 x=np.array(x), y=np.array(y), height=error)
             self.addItem(errbars)
-        if fit is not None:
+        if fit is not None and not self.mismatch['Fit values']:
             xi = np.argsort(x)
             self.plot(x[xi], fit[xi])
 
+    def length_warning(self):
+        text = "⚠️ dataset lengths mismatch:\n"
+        for key in self.mismatch:
+            if self.mismatch[key]:
+                text += key + ', '
```

Comment:
``", ".join()``
The output is otherwise also a bit strange with only one mismatch.